src/modules/did/did.service.ts

```diff
@@ -179,6 +179,37 @@ export class DIDService {
     }
   }
 
+  /**
+   * Refresh the DID Document cache for a given DID.
+   * Also retrieves all claims from IPFS for the document.
+   * @param {string} did
+   */
+  public async fullRefreshCachedDocument(did: string) {
+    try {
+      this.logger.info(
+        `refreshing cached document for did (full mode): ${did}`,
+      );
+      const cachedDIDDocument = await this.didRepository.findOne(did);
+
+      const logs = await this.getAllLogs(did);
+      const updatedDidDocument = this.resolveDocumentFromLogs(did, logs);
+      const updatedServices = await this.resolveNotCachedClaims(
+        updatedDidDocument.service,
+      );
+
+      const updatedEntity = DIDDocumentEntity.create({
+        ...cachedDIDDocument,
+        ...updatedDidDocument,
+        service: updatedServices,
+        logs: JSON.stringify(logs),
+      });
+
+      return this.didRepository.save(updatedEntity);
+    } catch (err) {
+      this.logger.error(err);
+    }
+  }
```

Comment:
I don't think it is necessary to retrieve the previously cached document (i.e. this line `const cachedDIDDocument = await this.didRepository.findOne(did);`) as the `id` of the entity is the `did` in question so if an entity exists, I think we can just replace it. If this is the case, then this new `fullRefreshCachedDocument` is essentially the same as `addCachedDocument`, so I think the two methods should be combined somehow.